src/main/kotlin/org/wa/rceditor/application/viewmodel/MainViewModel.kt

```diff
@@ -1,113 +1,320 @@
+            addRelation(it)
+        }
+
+        container.manifest.dublinCore.source.forEach {
+            addSource(it.identifier, it.language, it.version)
+        }
+
+        container.manifest.checking.checkingEntity.forEach {
+            addCheckingEntity(it)
+        }
+
+        container.manifest.projects.forEach {
+            addProject(it.title, it.versification, it.identifier, it.sort, it.path,
+                    if(it.categories.isNotEmpty()) it.categories.first() else "Unknown")
+        }
     }
 
-    fun handleOpenDocumentSelected() {
-        model.onOpenDocumentSelected()
+    fun saveResourceContainer() {
+        if (validateData()) {
+            container.manifest.dublinCore.conformsTo = conformsto
+            container.manifest.dublinCore.creator = creator
+            container.manifest.dublinCore.description = description
+            container.manifest.dublinCore.format = format
+            container.manifest.dublinCore.identifier = identifier
+            container.manifest.dublinCore.issued = issued.toString()
+            container.manifest.dublinCore.modified = modified.toString()
+            container.manifest.dublinCore.language.direction = languageDirection
+            container.manifest.dublinCore.language.identifier = languageIdentifier
+            container.manifest.dublinCore.language.title = languageTitle
+            container.manifest.dublinCore.publisher = publisher
+            container.manifest.dublinCore.rights = rights
+            container.manifest.dublinCore.subject = subject
+            container.manifest.dublinCore.title = title
+            container.manifest.dublinCore.type = type
+            container.manifest.dublinCore.version = version
+            container.manifest.checking.checkingLevel = checkingLevel
+
+            container.manifest.dublinCore.contributor = contributors.map { it.value }.toMutableList()
+            container.manifest.dublinCore.relation = relations.map { it.value }.toMutableList()
+            container.manifest.dublinCore.source = sources.map { it.toSource() }.toMutableList()
+            container.manifest.checking.checkingEntity = checkingEntities.map { it.value }.toList()
+            container.manifest.projects = projects.map { it.toProject() }.toList()
+
+            MainUseCase(RCManagerImpl()).saveResourceContainer(container)
+                    .observeOn(JavaFxScheduler.platform())
+                    .subscribeOn(Schedulers.io())
+                    /*.onErrorReturn {
+                        println(it.message)
+                    }*/
+                    .subscribe {
+                        println("Saved")
+                    }
+
+            // TODO show success popup
+        }
+        else {
+            // TODO show error popup
+        }
     }
 
-    fun handleSaveDocumentSelected() {
-        commit()
-        model.onSaveDocumentSelected()
+    fun validateData(): Boolean {
```

Comment:
If you haven't already, I would move this to a `ValidateManifest` use case or similarly named

Reply:
@mbr4477 But I'm validating viewmodel properties here. And domain layer can't have an access to the presentation layer properties, can it?

Reply:
I see. I still like the idea of the validation happening in the domain layer, more isolated from the view/viewmodel. Then you could theoretically reuse that validation logic if you wanted to validate the fields from an RC loaded from the disk. One approach might be to have a `ValidateResourceContainer` use case that has public methods like `fun validateVersion(version: String): Boolean` that you can call from the view mode.

Reply:
@mbr4477 Got it, I'll do it